test/DCNTToken.test.ts

```diff
@@ -89,65 +77,36 @@ describe("DCNTToken", async function () {
+    });
 
-      describe("Depending on the amount", function () {
-        let maxToMint: BigNumber;
-
-        beforeEach(async function () {
-          // calculate the max amount of tokens that can be minted
-          maxToMint = originalTotalSupply.mul(mintCapBPs).div(10000);
-        });
-
-        describe("Max mint amount", function () {
-          it("Should allow the owner to mint more tokens", async function () {
-            await dcnt.mint(owner.address, maxToMint);
-            expect(await dcnt.totalSupply()).to.eq(
-              originalTotalSupply.add(maxToMint)
-            );
-          });
-        });
-
-        describe("Over max mint amount", function () {
-          let overMaxToMint: BigNumber;
-
-          beforeEach(function () {
-            overMaxToMint = maxToMint.add(1);
-          });
+    describe("Updating the MintAuthorization contract", function () {
+      let newInstance: UnlimitedMint;
 
-          it("Should not allow the owner to mint more tokens", async function () {
-            await expect(
-              dcnt.mint(owner.address, overMaxToMint)
-            ).to.be.revertedWith("MintExceedsMaximum()");
-          });
-        });
+      beforeEach(async function () {
+        newInstance = await new UnlimitedMint__factory(owner).deploy();
       });
 
-      describe("Depending on the time", function () {
-        beforeEach(async function () {
-          // dummy mint to set the timeout interval
-          await dcnt.mint(owner.address, 0);
-        });
+      it("Should allow owner to update to a new instance", async function () {
+        await dcnt.updateMintAuthorization(newInstance.address);
+        expect(await dcnt.mintAuthorization()).to.eq(newInstance.address);
+      });
 
-        it("Should not allow the owner to mint after having just minted", async function () {
-          await expect(dcnt.mint(owner.address, 1)).to.be.revertedWith(
-            "MintTooSoon()"
-          );
-        });
+      it("Should not allow non-owner to update to a new instance", async function () {
+        const updateMintAuthorizationRole = ethers.utils.keccak256(ethers.utils.toUtf8Bytes('UPDATE_MINT_AUTHORIZATION_ROLE'));
+        const originalMintAuthorizationAddress = await dcnt.mintAuthorization();
 
-        it("Should allow the owner to mint after the minimum mint interval", async function () {
-          await time.increase(minimumMintInterval);
-          const toMint = 1;
-          await dcnt.mint(owner.address, toMint);
-          expect(await dcnt.totalSupply()).to.eq(
-            originalTotalSupply.add(toMint)
-          );
-        });
+        await expect(
+          dcnt.connect(nonOwner).updateMintAuthorization(newInstance.address)
+        ).to.be.revertedWith(`AccessControl: account ${nonOwner.address.toLowerCase()} is missing role ${updateMintAuthorizationRole}`);
+        expect(await dcnt.mintAuthorization()).to.eq(originalMintAuthorizationAddress);
       });
-    });
+    })
```

Comment:
tests to revoke the 2 roles, then try to update the authorization or mint new tokens would be good here